src/Hl7.Fhir.Specification.Tests/Validation/ValidationManifestTest.cs

```diff
@@ -0,0 +1,256 @@
+﻿using Hl7.Fhir.Model;
+using Hl7.Fhir.Serialization;
+using Hl7.Fhir.Specification.Source;
+using Hl7.Fhir.Validation;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Newtonsoft.Json.Converters;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Xml;
+
+namespace Hl7.Fhir.Specification.Tests
+{
+    [TestClass]
+    public class ValidationManifestTest
+    {
+        private static Validator _testValidator;
+        private static DirectorySource _dirSource;        
+
+        [ClassInitialize]
+        public static void ClassInitialize(TestContext context)
+        {
+            _dirSource = new DirectorySource(@"TestData\\validation-test-suite", new DirectorySourceSettings { IncludeSubDirectories = true });
+            var zipSource = ZipSource.CreateValidationSource();
+            var testResolver = new CachedResolver(new MultiResolver(zipSource, _dirSource));
+
+            var settings = ValidationSettings.CreateDefault();
+            settings.GenerateSnapshot = true;
+            settings.GenerateSnapshotSettings = new Snapshot.SnapshotGeneratorSettings()
+                                                {
+                                                    ForceRegenerateSnapshots = true,
+                                                    GenerateSnapshotForExternalProfiles = true,
+                                                    GenerateElementIds = true
+                                                };
+            settings.ResourceResolver = testResolver;
+
+           _testValidator = new Validator(settings);
+        }
+
+        [Ignore]
+        [TestMethod]
+        [DataTestMethod]
+        [CustomDataSource]
+        public void TestValidationManifest(ValidationTestCase testCase)
+        {
+            RunTestCase(testCase);
+        } 
+
+        public static void RunTestCase(ValidationTestCase testCase)
+        {
+            Resource testResource = null;
+            if (testCase.FileName.EndsWith(".xml"))
+            {
+                var reader = XmlReader.Create(@"TestData\\validation-test-suite\\" + testCase.FileName);
```

Comment:
You could use also a textreader (which you do at Json as well), then you can write it shorter:

 ```c#
            var resourceText = File.ReadAllText(@$"TestData\validation-test-suite\{testCase.FileName}");
            var testResource = testCase.FileName.EndsWith(".xml") ?
                new FhirXmlParser().Parse<Resource>(resourceText) :
                new FhirJsonParser().Parse<Resource>(resourceText);
```